src/lib/core/overlay/position/viewport-ruler.ts

```diff
@@ -54,10 +62,16 @@ export class ViewportRuler {
 
     return {top, left};
   }
+
+  /** Caches the latest client rectangle of the document element. */
+  _cacheViewportGeometry?() {
+    this._documentRect = document.documentElement.getBoundingClientRect();
+  }
+
 }
 
 export function VIEWPORT_RULER_PROVIDER_FACTORY(parentDispatcher: ViewportRuler) {
-  return parentDispatcher || new ViewportRuler();
+  return parentDispatcher || new ViewportRuler(new ScrollDispatcher());
```

Comment:
Instead of doing `new ScrollDispatcher()`, it should inject the existing scroll dispatcher. 
(see `MdIconRegistry` for an example)